text/0000-partial_types.md

```diff
@@ -0,0 +1,672 @@
+
+As we see, 
+ - `%empty : %{%deny self.*}` or `%empty : %{}` integrity
+ - `%full  : %{%fit  self.*}` or `%full  : %{self.*}` integrity
+
+
+## Partial parameters
+
+We add enough integrity, and could write partial parameters for non-virtual function declarations:
+```rust
+// case (D1)
+fn re_ref_t (& p : & %{self.t, %ignore self._} Point) -> &f64 {
+   &p.t
+}
+
+// case (D2)
+fn refmut_w (&mut p : &mut %{self.w, %ignore self._} Point) -> &mut f64 {
+   &mut p.w
+}
+```
+
+Where :
+ - `%ignore` is a "don't care which exactly" quasi filed-integrity (`%_` is a whole type integrity and it is unclear if we could use it in both contents)
+
+But `%ignore self._` quasi-filed-integrity of quasi-field looks annoying, so we simplify a bit adding `%any : %ignore self._`.
+
+```rust
+// case (D3)
+// FROM case (D1)
+fn re_ref_t (& p : & %{self.t, %any} Point) -> &f64 {
+   &p.t
+}
+
+// case (D4)
+// FROM case (D2)
+fn refmut_w (&mut p : &mut %{self.w, %any} Point) -> &mut f64 {
+   &mut p.w
+}
+
+// case (D5)
+struct PointExtra {
+    x: f64,
+    y: f64,
+    saved_x: f64,
+    saved_y: f64,
+}
+
+fn x_store(&mut p1 : &mut %{self.saved_x, %any} PointExtra, & p2 : & %{self.x, %any} PointExtra) {
+    *p1.saved_x = *p2.x
+}
+
+fn x_restore(&mut p1 : &mut %{self.x, %any} PointExtra, & p2 : & %{self.saved_x, %any} PointExtra) {
+    *p1.x = *p2.saved_x;
+}
+```
+
+or use `where` clause if integrity is extra verbose:
+```rust
+// case (D6)
+// FROM case (D5)
+
+fn x_store(&mut p1 : &mut %fit_sv_x PointExtra, & p2 : & %fit_x PointExtra) 
+    where %fit_sv_x : %{self.saved_x, %any},
+          %fit_x : %{self.x, %any}
+{
+    *p1.saved_x = *p2.x
+}
+
+fn x_restore(&mut p1 : &mut %fit_x PointExtra, & p2 : & %fit_sv_x PointExtra) 
+    where %fit_sv_x : %{self.saved_x, %any},
+          %fit_x : %{self.x, %any}
+{
+    *p1.x = *p2.saved_x;
+}
+```
+
+Implementation parameters are mostly same:
+```rust
+// case (D7)
+impl Point {
+    pub fn x_refmut(&mut self : &mut %{self.x, %any} Self) -> &mut f64 {
+        &mut self.x
+    }
+
+    pub fn y_refmut(&mut self : &mut %{self.y, %any} Self) -> &mut f64 {
+        &mut self.y
+    }
+}
+```
+
+We could also use multiple sub-parameters of same parameter
+```rust
+// case (D8)
+    pub fn xy_swich(&mut self : &mut %{self.{x, y}, %any} Self) {
+        let tmp = *self.x;
+        *self.x = *self.y;
+        *self.y = tmp;
+    }
+```
+
+Now type integrity guarantee to compiler, that only some fields has an access inside function, but not the rest of them.
+So, no extra lock on `self` is needed, only for `%fit` fields.
+
+Now compiler can catch "out of scope parameter" errors
+```rust
+// case (D9)
+    pub fn xt_refmut(&self : &mut %{self.xt, %any} Self) -> &mut f64 {
+        //                               ^~~~~~
+        // error: no field 'self.xt' on type `self`
+        &mut self.xt
+    }
+```
+
+Since using `%ignore` filed is **unsafe**, trying to use ignoring field is a compile error:
```

Comment:
The RFC says nothing about `%ignore` being unsafe before this point.

Reply:
Thank you! I'll try to simplify RFC a lot, using your review. 
It is 4rth version, and some old thoughts looks strange now

Reply:
I've changed to "Since using %ignore filed is unsafe by type (we have no guarantee, that some field is permitted), trying to use ignoring field is a compile error"
Is it clearer now?